spark/v3.5/spark/src/main/java/org/apache/iceberg/spark/SparkConfParser.java

```diff
@@ -227,7 +225,26 @@ public ThisT tableProperty(String name) {
       return self();
     }
 
-    protected T parse(Function<String, T> conversion, T defaultValue) {
+    public ThisT check(Predicate<T> newCheck, String newErrMsg) {
+      this.check = newCheck;
+      this.errMsg = newErrMsg;
+      return self();
+    }
+
+    protected T parseAndCheck(Function<String, T> conversion, T defaultValue) {
```

Comment:
Is it possible for `ProcedureInput` to utilize this method as well?

Reply:
This one depends on some state, I'll have to take a look `ProcedureInput`.